### DISCUSSION TITLE
Expand disk size of OS drive step is failing in Initialize-VM.ps1

### DISCUSSION THREAD

**Discussion by Malaw80s** [actions/virtual-environments#5432] [General questions] [answered]
Hi There,

My scrip is failing at the Initialize-VM.ps1 for Windows 2022

It's failing at this point of the code of Initialize-VM.ps1:
 Expand disk size of OS drive
 $driveLetter = "C"
 $size = Get-PartitionSupportedSize -DriveLetter $driveLetter
 Resize-Partition -DriveLetter $driveLetter -Size $size.SizeMax
 Get-Volume | Select-Object DriveLetter, SizeRemaining, Size | Sort-Object DriveLetter

What does this portion of the code do and can I comment it out? I have retried running it multiple times but always fails at this point :( 

Error:
==> vhd: Provisioning with powershell script: C:\virtual-environments\images\win/scripts/Installers/Install-Choco.ps1
vhd: Set TLS1.2
vhd: Install chocolatey
vhd: WARNING: 'choco' was found at 'C:\ProgramData\Chocolatey\bin\choco.exe'.
vhd: WARNING: An existing Chocolatey installation was detected. Installation will not continue.
vhd: For security reasons, this script will not overwrite existing installations.
vhd:
vhd: Please use choco upgrade chocolatey to handle upgrades of Chocolatey itself.
vhd: Chocolatey v1.1.0
vhd: 2 validations performed. 1 success(es), 1 warning(s), and 0 error(s).
vhd:
vhd: Validation Warnings:
vhd: - A pending system reboot request has been detected, however, this is
vhd: being ignored due to the current command being used 'feature'.
vhd: It is recommended that you reboot at your earliest convenience.
vhd:
vhd: Nothing to change. Config already set.
vhd: C:\tools
==> vhd: Provisioning with powershell script: C:\virtual-environments\images\win/scripts/Installers/Initialize-VM.ps1
vhd: Disable Server Manager on Logon
vhd:
vhd: TaskPath TaskName State
vhd: -------- -------- -----
vhd: \Microsoft\Windows\Server Manager\ ServerManager Disabled
vhd: Disable 'Allow your PC to be discoverable by other PCs' popup
vhd:
vhd: Property : {}
vhd: PSPath : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Network\NewNetw
vhd: orkWindowOff
vhd: PSParentPath : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Network
vhd: PSChildName : NewNetworkWindowOff
vhd: PSDrive : HKLM
vhd: PSProvider : Microsoft.PowerShell.Core\Registry
vhd: PSIsContainer : True
vhd: SubKeyCount : 0
vhd: View : Default
vhd: Handle : Microsoft.Win32.SafeHandles.SafeRegistryHandle
vhd: ValueCount : 0
vhd: Name : HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Network\NewNetworkWindowOff
vhd:
vhd: Disable Windows Update
vhd: Disabled Windows Update
vhd: Disable UAC
vhd: User Access Control (UAC) has been disabled.
vhd: Disable IE Welcome Screen
vhd:
vhd: Property : {(default)}
vhd: PSPath : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet
vhd: Explorer\Main
vhd: PSParentPath : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet Explorer
vhd: PSChildName : Main
vhd: PSDrive : HKLM
vhd: PSProvider : Microsoft.PowerShell.Core\Registry
vhd: PSIsContainer : True
vhd: SubKeyCount : 0
vhd: View : Default
vhd: Handle : Microsoft.Win32.SafeHandles.SafeRegistryHandle
vhd: ValueCount : 1
vhd: Name : HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet Explorer\Main
vhd:
vhd: Disabled IE Welcome screen
vhd: Disable IE ESC
vhd: IE Enhanced Security Configuration (ESC) has been disabled.
vhd: Setting local execution policy
vhd:
vhd: Scope : MachinePolicy
vhd: ExecutionPolicy : Undefined
vhd:
vhd:
vhd: Scope : UserPolicy
vhd: ExecutionPolicy : Undefined
vhd:
vhd:
vhd: Scope : Process
vhd: ExecutionPolicy : Unrestricted
vhd:
vhd:
vhd: Scope : CurrentUser
vhd: ExecutionPolicy : Undefined
vhd:
vhd:
vhd: Scope : LocalMachine
vhd: ExecutionPolicy : Unrestricted
vhd:
vhd: Enable long path behavior
==> vhd: Resize-Partition : Size Not Supported
==> vhd:
==> vhd: Extended information:
==> vhd: The partition is already the requested size.
==> vhd:
==> vhd: Activity ID: {c64671f8-efed-4898-90ee-561f61d08e5a}
==> vhd: At C:\Windows\Temp\script-625e3290-eac0-0e84-1dfd-29fc0a1005ef.ps1:75 char:1
==> vhd: + Resize-Partition -DriveLetter $driveLetter -Size $size.SizeMax
==> vhd: + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd: + CategoryInfo : NotSpecified: (StorageWMI:ROOT/Microsoft/.../MSFT_Partition) [Resize-Partition], CimExce
==> vhd: ption
==> vhd: + FullyQualifiedErrorId : StorageWMI 4097,Resize-Partition
==> vhd:
==> vhd: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?

**Comment by miketimofeev** [comment, marked as answer]
@Malaw80s what VM size do you use? You can try removing this line from the script:
https://github.com/actions/virtual-environments/blob/d6ddbcd278fd03cec014d3596dce618cca969046/images/win/scripts/Installers/Initialize-VM.ps1#L72-L76

**Reply by Malaw80s** [reply]
Hi @miketimofeev ,

I'm using the default size set to "vm_size": "Standard_D8s_v4" that is set under the windows2022.json, should I pick another vm size?

after commenting on the code you mentioned above, I'm running to the error below:

==> vhd: Provisioning with powershell script: C:\virtual-environments\images\win/scripts/Installers/Install-PowerShellModules.ps1
    vhd: Setup PowerShellGet
    vhd:
    vhd: Name                           Version          Source           Summary
    vhd: ----                           -------          ------           -------
    vhd: nuget                          2.8.5.208        https://onege... NuGet provider for the OneGet meta-package manager
    vhd: Installing DockerMsftProvider module
    vhd: Installing MarkdownPS module
    vhd: Installing Pester module
    vhd: WARNING: The version '5.3.2' of module 'Pester' is currently in use. Retry the operation after closing the
    vhd: applications.
    vhd: Installing PowerShellGet module
    vhd: WARNING: The version '1.4.7' of module 'PackageManagement' is currently in use. Retry the operation after closing the
    vhd: applications.
    vhd: Installing PSScriptAnalyzer module
    vhd: Installing PSWindowsUpdate module
    vhd: Installing SqlServer module
    vhd: Installing VSSetup module
    vhd: Installing Microsoft.Graph module
    vhd: Installing AWSPowershell module
==> vhd: PackageManagement\Install-Package : Package 'AWSPowerShell' failed to be installed because: There is not enough space
==> vhd: on the disk.
==> vhd: At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\2.2.5\PSModule.psm1:9711 char:34
==> vhd: + ... talledPackages = PackageManagement\Install-Package @PSBoundParameters
==> vhd: +                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd:     + CategoryInfo          : InvalidResult: (AWSPowerShell:String) [Install-Package], Exception
==> vhd:     + FullyQualifiedErrorId : Package '{0}' failed to be installed because: {1},Microsoft.PowerShell.PackageManagement
==> vhd:    .Cmdlets.InstallPackage
==> vhd:
==> vhd: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?

**Reply by miketimofeev** [reply]
the size is ok. What Azure region do you use for the image generation?

**Reply by Malaw80s** [reply]
Hi @miketimofeev ,

I'm using "East US2"

**Reply by Malaw80s** [reply]
Hi @miketimofeev
I tried changing it to "East US", and even increased the disk size to "os_disk_size_gb": "512" in the windows2022.json file
but still fails at this point:

==> vhd: PackageManagement\Install-Package : Package 'AWSPowerShell' failed to be installed because: There is not enough space
==> vhd: on the disk.
==> vhd: At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\2.2.5\PSModule.psm1:9711 char:34
==> vhd: + ... talledPackages = PackageManagement\Install-Package @PSBoundParameters
==> vhd: +                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd:     + CategoryInfo          : InvalidResult: (AWSPowerShell:String) [Install-Package], Exception
==> vhd:     + FullyQualifiedErrorId : Package '{0}' failed to be installed because: {1},Microsoft.PowerShell.PackageManagement
==> vhd:    .Cmdlets.InstallPackage
==> vhd:
==> vhd: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?

**Reply by miketimofeev** [reply]
@Malaw80s well then we definitely need those removed lines but it's still unclear why Resize-Partition doesn't work for you

**Reply by Malaw80s** [reply]
Hi @miketimofeev 

I see the error above even with the code below commented out:
```
# Expand disk size of OS drive
# $driveLetter = "C"
# $size = Get-PartitionSupportedSize -DriveLetter $driveLetter
# Resize-Partition -DriveLetter $driveLetter -Size $size.SizeMax
# Get-Volume | Select-Object DriveLetter, SizeRemaining, Size | Sort-Object DriveLetter
```

any thoughts?

**Reply by miketimofeev** [reply]
yes, we can't comment it out definitely.
Could you try building the image using the latest main and West US2 region? Also, do you see any issues with windows-2019?

**Reply by Malaw80s** [reply]
@miketimofeev 
Yes, I'll test it out with the latest main and West US2.
Yes, I ran into this issue with windows-2019 when I first started this process, but I can check again.

**Reply by Malaw80s** [reply]
Hi @miketimofeev 

I got the latest code from main with no code commented out, and ran the script with "West US2" and it worked with no issues 👍 

Thank you so much for your help :)

**Comment by mxa0079** [comment]
I am having this problem running in South Central US in the latest version of main.

**Reply by mxa0079** [reply]
I also tried running the script with "West US2" and I am still getting the following error:

```
   vhd: PSIsContainer : True
    vhd: SubKeyCount   : 0
    vhd: View          : Default
    vhd: Handle        : Microsoft.Win32.SafeHandles.SafeRegistryHandle
    vhd: ValueCount    : 1
    vhd: Name          : HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet Explorer\Main
    vhd:
    vhd: Disabled IE Welcome screen
    vhd: Disable IE ESC
    vhd: IE Enhanced Security Configuration (ESC) has been disabled.
    vhd: Setting local execution policy
    vhd:
    vhd: Scope           : MachinePolicy
    vhd: ExecutionPolicy : Undefined
    vhd:
    vhd:
    vhd: Scope           : UserPolicy
    vhd: ExecutionPolicy : Undefined
    vhd:
    vhd:
    vhd: Scope           : Process
    vhd: ExecutionPolicy : Unrestricted
    vhd:
    vhd:
    vhd: Scope           : CurrentUser
    vhd: ExecutionPolicy : RemoteSigned
    vhd:
    vhd:
    vhd: Scope           : LocalMachine
    vhd: ExecutionPolicy : Unrestricted
    vhd:
    vhd: Enable long path behavior
==> vhd: Resize-Partition : Size Not Supported
==> vhd:
==> vhd: Extended information:
==> vhd: The partition is already the requested size.
==> vhd:
==> vhd: Activity ID: {ba5fe9f8-13c3-4ccf-8978-6a746d41a4a1}
==> vhd: At C:\Windows\Temp\script-63a481cd-ed28-d145-99d9-075edeea8873.ps1:75 char:2
==> vhd: +  Resize-Partition -DriveLetter $driveLetter -Size $size.SizeMax
==> vhd: +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd:     + CategoryInfo          : NotSpecified: (StorageWMI:ROOT/Microsoft/.../MSFT_Partition) [Resize-Partition], CimExce
==> vhd:    ption
==> vhd:     + FullyQualifiedErrorId : StorageWMI 4097,Resize-Partition
==> vhd:
==> vhd: Script exited with non-zero exit status: 1. Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?

```

**Reply by Malaw80s** [reply]
All these issues went away once I ran the pipeline using Microsoft Hosted Agent as an agent pool

**Comment by pebroz** [comment]
Same issue in Norway East. I'd wish for better logic in the "# Expand disk size of OS drive" step in Initialize-VM.ps1

**Comment by rwilkett** [comment]
I am having the same issue with East US.
==> vhd: Resize-Partition : Size Not Supported
==> vhd:
==> vhd: Extended information:
==> vhd: The partition is already the requested size.
==> vhd:
==> vhd: Activity ID: {2d586d4a-2dce-4538-944a-7f7835e168fc}
==> vhd: At C:\Windows\Temp\script-63d849cf-9c12-df5b-d4c7-0d6abd4cd709.ps1:75 char:1
==> vhd: + Resize-Partition -DriveLetter $driveLetter -Size $size.SizeMax
==> vhd: + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd:     + CategoryInfo          : NotSpecified: (StorageWMI:ROOT/Microsoft/.../MSFT_Partition) [Resize-Partition], CimExce
==> vhd:    ption
==> vhd:     + FullyQualifiedErrorId : StorageWMI 4097,Resize-Partition
==> vhd:
==> vhd: Script exited with non-zero exit status: 1. Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)? c
==> vhd: Provisioning step had errors: Running the cleanup provisioner, if present...
